tools/gfx/vulkan/render-vk.cpp

```diff
@@ -3788,17 +3800,6 @@ Result VKDevice::PipelineCommandEncoder::bindRootShaderObjectImpl(
         0,
         nullptr);
 
-    // Bind Push Constants.
-    for (auto pushConstantRange : specializedLayout->m_pushConstantRanges)
-    {
-        m_device->m_api.vkCmdPushConstants(
-            m_commandBuffer->m_commandBuffer,
-            specializedLayout->m_pipelineLayout,
-            pushConstantRange.stageFlags,
-            pushConstantRange.offset,
-            pushConstantRange.size,
-            bindState.pushConstantBuffer.getBuffer() + pushConstantRange.offset);
-    }
```

Comment:
It is good to have this move out of the `bindState`, and the next step would seemingly (eventually) be to move the `vkUpdateDescriptorSet` and `vkCmdBindDescriptorSets` calls out as well.

Realistically, further cleanup can/should probably wait until we start to implement the more advanced design for root shader objects where the application cannot arbitrarily persist them, and then behave closer to the command encoders in practice (created for short-term use and then dropped).